### ISSUE TITLE
Use case "verify all classes used as entity ids"

### ISSUE THREAD

**Issue by dpolivaev** [jqno/equalsverifier#271]
Just sharing

https://gist.github.com/dpolivaev/5458f51b51770a880b34be045a1a35a2

Here I use combination of reflections library with equalsverifier to automatically check that all classes used as hibernate entity IDs have correct implementations of hash and equals.

Instead of used https://github.com/ronmamo/reflections one could also use Junit5 ReflectionSupport, I am going to check it later. The last could also be used to check implementations of all equals and hash methods implemented in a given package. Probably some suppresses warnings could be configured by annotations either on the methods  or on the checked class. But as it is now it is already useful.

**Comment by jqno** [comment]
I've thought of this before, but could never figure out how to do it properly. Your approach looks nice, especially if we combine it with `configure()`, I think. I'll look into it!

**Comment by jqno** [comment]
I've just released a new version (3.2) with this feature. It adds `forPackage` and `forClasses` methods. Please let me know what you think!

**Comment by dpolivaev** [comment]
To be honest I do not understand how methods `forPackage` and `forClasses` can be used to check hibernate entities.

**Comment by jqno** [comment]
I suppose I misunderstood your initial request -- I thought you were looking for a way to test multiple classes at once, I didn't realise you wanted to be more specific.

**Comment by jqno** [comment]
I can add a `forClasses(List<Class<?>>)` overload, so you can easily put in a list of classes that you found using the Reflections library. I'm not keen to embed such a think inside EqualsVerifier though.

**Comment by dpolivaev** [comment]
You decide as you like, I do not mind.

**Comment by dpolivaev** [comment]
Maybe adding some user defined filter like in a stream API could be used so that verification could be applied only to selected classes, for example for classes with some particular annotation. However, my solution is still easier to use because it can be applied to all classes in any subpackages of the base package.

**Comment by jqno** [comment]
If I add a `forClasses` overload that accepts an Interator, you could replace everything from line 46 and down in https://gist.github.com/dpolivaev/5458f51b51770a880b34be045a1a35a2 with the following code:

```
EqualsVerifier.forClasses(checkedClasses)
        .usingGetClass()
        .suppress(Warning.NONFINAL_FIELDS)
        .verify();
```

That should already be helpful, no? `.verify()` on `.forClasses()` already aggregates the content of the `report()` in a way very similar to what you do in your code.

**Comment by jqno** [comment]
I've just released version 3.3.

It contains the `forClasses` overload discussed above. Also, there's an `except()` method that expects a `Predicate`, that may be of help in your use case. For example, you could do something like this:

```
EqualsVerifier.forPackage("some.package")
    .except(c -> c.getAnnotation(Id.class) == null)
    .usingGetClass()
    .suppress(Warning.NONFINAL_FIELDS)
    .verify();
```

See https://github.com/jqno/equalsverifier/pull/307 for more info on that.
